server/src/main/java/org/elasticsearch/common/settings/SecureSettingsLoader.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.settings;
+
+import joptsimple.OptionSet;
+import joptsimple.OptionSpec;
+
+import org.elasticsearch.cli.Command;
+import org.elasticsearch.cli.ProcessInfo;
+import org.elasticsearch.cli.Terminal;
+import org.elasticsearch.env.Environment;
+
+import java.util.Optional;
+
+/**
+ * An interface for implementing {@link SecureSettings} loaders, that is, implementations that create, initialize and load
+ * secrets stores.
+ */
+public interface SecureSettingsLoader {
+    LoadedSecrets load(
+        Environment environment,
+        Terminal terminal,
+        ProcessInfo processInfo,
+        OptionSet options,
+        Command autoConfigureCommand,
```

Comment:
Auto configuration and the enrollment token seem like something only tangentially related to secure settings (since auto configuration may add some secure settings).

Instead of having the loader do both, could we instead have a similar pattern as before, where we (1) load secure settings, (2) call auto configuration and (3) possibly reload secure settings?

Reply:
OK, sounds good. Let me see how I can abstract that into a separate concept.